### ISSUE TITLE
Dirichlet example fails with any a <= 1

### ISSUE THREAD

**Issue by tdihp** [pymc-devs/pymc#692]
I'm using pymc3 master, using pymc3/examples/dirichlet.py code, but when I set a <= 1.0, sampler just give nan.

Here's how to reproduce, change `a = 1.0` to any larger numbers will work just fine

```
import pymc3 as pm
import numpy as np
model = pm.Model()
with model:
    k = 6
    a = 1.0
    p, p_m1 = model.TransformedVar(
        'p', pm.Dirichlet.dist(np.repeat(a, k), shape=k),
        pm.simplextransform)
with model:
    step = pm.Metropolis()
    trace = pm.sample(1000, step)

result = trace['p']
print result
```

I suppose it's something to do with `Dirichlet.mode`, which will be all nan if any(a <= 1).

So how can I get the sampler work?

**Comment by kiudee** [comment]
You are correct, the cause here is the mode. The problem is that the sampling starts with the test values as initial point, which in the case of free random variables is set to:

``` python
self.tag.test_value = np.ones(distribution.shape, distribution.dtype) * distribution.default()
```

The only default for discrete random variables is the mode, which is why it’s using the `nan` vector.

Is there any reason why we only use the mode for discrete variables? If not, I would propose adding the mean and median to the defaults and to check `np.all(np.isfinite(v))` when checking the defaults.

PS: I just realized that the Dirichlet is a continuous distribution for which it tries the mean first. The mean should not be `nan` here. I will look into it.

**Comment by kiudee** [comment]
Update: It is not the mode. The `nan` array is caused by this:

``` python
if isinstance(val, t.TensorVariable):
    return val.tag.test_value
```

The test value for the mean is `array(nan)`.

**Comment by kiudee** [comment]
A first workaround would be to give an explicit test value:

``` python
params = np.repeat(a, k)
p, p_m1 = model.TransformedVar('p',
    pm.Dirichlet.dist(a=params, testval=params/np.sum(params), shape=k),
    pm.simplextransform)
```

**Comment by kiudee** [comment]
Using the workaround and the most recent version of pymc3 with fix #693, your example code works.
As soon as I have more time, I will look into the issue with the test value.

**Comment by tdihp** [comment]
Thanks!
Didn't know there is testval before.
